src/parser/Parser.spec.ts

```diff
@@ -139,6 +139,88 @@ describe('parser', () => {
         });
     });
 
+    describe('optional chaining operator', () => {
```

Comment:
Should we have something in there that validates it doesn't match something like:

```
?["test"]
```
It should be some sort of print. Not sure what test area that would go in.

Reply:
Oops. Good catch. Yeah, I need to tweak the lexer to handle this situation. I need to look to the left of the question mark, and if it's preceeded by only whitespace and then a newline or colons, then the `?` should be its own token, separate from the trailing `[` or `(`

Reply:
I fixed it in d317171a9c18c9f35bbda5592f56c30e3ff8a399